File: backend/internal/model/post_template.go
package model

import "time"

// PostTemplate は投稿テンプレートのモデル。
// よく使う投稿フォーマットを保存し、投稿作成時に再利用できる。
type PostTemplate struct {
	ID              uint      `json:"id"`
	UserID          uint      `json:"user_id"`
	Name            string    `json:"name"`
	TitleTemplate   string    `json:"title_template"`
	ContentTemplate string    `json:"content_template"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: frontend/src/types/postTemplate.ts
export interface PostTemplate {
  id: number;
  user_id: number;
  name: string;
  title_template: string;
  content_template: string;
  created_at: string;
  updated_at: string;
}

export interface CreatePostTemplateRequest {
  name: string;
  title_template?: string;
  content_template: string;
}

export interface UpdatePostTemplateRequest {
  name?: string;
  title_template?: string;
  content_template?: string;
}

export interface PostTemplateListResponse {
  templates: PostTemplate[];
  total: number;
  limit: number;
  offset: number;
}
